### ISSUE TITLE
LocalReadFile::preadv returning EINVAL

### ISSUE THREAD

**Issue by acvictor** [facebookincubator/velox#6648]
### Bug description

With LocalReadFile::preadv, when the size of iovecs grows beyond `IOV_MAX` (1024 in the case of Linux and macOS), it results in the following exception where bytes read is returned as -1 and `errno` is set to `EINVAL`.

```
2023-09-13 16:57:50,332 ERROR TaskResources [Executor task launch worker for task 49.0 in stage 35.0 (TID 1776)]: Task 1776 failed by error: 
java.lang.RuntimeException: Exception: VeloxException
Error Source: RUNTIME
Error Code: UNKNOWN
Reason: [Range Constraint Violation : 18446744073709551615==25430225] : Should read exactly as requested. File name: /tpcds1tb/store_sales/ss_sold_date_sk=2452219/part-00117-7b0d8ab7-ae80-439f-a778-68924e57a6b7.c000.snappy.parquet, offset: 29447802, length: 25430225, read: 18446744073709551615
Retriable: False
Expression: size == bufferSize
Context: Split [Hive: file:/tpcds1tb/store_sales/ss_sold_date_sk=2452219/part-00117-7b0d8ab7-ae80-439f-a778-68924e57a6b7.c000.snappy.parquet 0 - 141667339] Task Gluten stage-35 task-1776
Top-Level Context: Same as context.
Function: read
File: /src/Velox/velox/dwio/common/InputStream.cpp
Line: 101
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 2  _ZN8facebook5velox4dwio6common9exception15LoggedExceptionC1EPKcmS6_S6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_SE_
# 3  _ZN8facebook5velox4dwio6common19ReadFileInputStream4readERKSt6vectorIN5folly5RangeIPcEESaIS8_EEmNS2_10MetricsLog11MetricsTypeE.cold
# 4  _ZN8facebook5velox4dwio6common11InputStream5vreadERKSt6vectorIPvSaIS5_EERKS4_INS2_6RegionESaISA_EENS2_10MetricsLog11MetricsTypeE
# 5  _ZN8facebook5velox4dwio6common13BufferedInput4loadENS2_10MetricsLog11MetricsTypeE
# 6  _ZN8facebook5velox7parquet10ReaderBase17scheduleRowGroupsERKSt6vectorIjSaIjEEiRNS1_18StructColumnReaderE
# 7  _ZN8facebook5velox7parquet16ParquetRowReader21advanceToNextRowGroupEv
# 8  _ZN8facebook5velox7parquet16ParquetRowReaderC2ERKSt10shared_ptrINS1_10ReaderBaseEERKNS0_4dwio6common16RowReaderOptionsEb
# 9  _ZNK8facebook5velox7parquet13ParquetReader15createRowReaderERKNS0_4dwio6common16RowReaderOptionsE
# 10 _ZN8facebook5velox9connector4hive14HiveDataSource8addSplitESt10shared_ptrINS1_14ConnectorSplitEE
# 11 _ZN8facebook5velox4exec9TableScan9getOutputEv
# 12 _ZN8facebook5velox4exec6Driver11runInternalERSt10shared_ptrIS2_ERS3_INS1_13BlockingStateEERS3_INS0_9RowVectorEE
# 13 _ZN8facebook5velox4exec6Driver4nextERSt10shared_ptrINS1_13BlockingStateEE
# 14 _ZN8facebook5velox4exec4Task4nextEPN5folly10SemiFutureINS3_4UnitEEE
# 15 _ZN6gluten24WholeStageResultIterator4nextEv
# 16 Java_io_glutenproject_vectorized_ColumnarBatchOutIterator_nativeHasNext
# 17 0x00007fd5fd5c8607
```
According to the [Linux man page](https://linux.die.net/man/2/preadv), `EINVAL` is returned when *the vector count iovcnt is less than zero or greater than the permitted maximum* and the limit is obtained as *IOV_MAX in <limits.h>*.

This failure is reproducible by running most TPCDS queries on a scale factor of 1000 and 100. I will raise a PR to fix the issue by batching calls to preadv when `iovecs.size() >= IOV_MAX`.

### System information

Velox System Info v0.0.2
Commit: d3f5dc09bb927c1cd992363baf50d196b034403d
CMake Version: 3.16.3
System: Linux-5.15.0-83-generic
Arch: x86_64
C++ Compiler: /usr/bin/c++
C++ Compiler Version: 9.4.0
C Compiler: /usr/bin/cc
C Compiler Version: 9.4.0
CMake Prefix Path: /usr/local;/usr;/;/usr;/usr/local;/usr/X11R6;/usr/pkg;/opt

**Comment by yingsu00** [comment]
@acvictor Will you be able to provide a repro test case?

**Comment by yingsu00** [comment]
@acvictor Is this issue already fixed by https://github.com/facebookincubator/velox/pull/6649? If yes can we close this issue?

**Comment by acvictor** [comment]
The issue has been fixed in #6649
